digicampipe/utils/utils.py

```diff
@@ -1,9 +1,28 @@
 import numpy as np
 import scipy.ndimage as ndimage
-
+from scipy.interpolate import interp1d
 # Define the integration function
 
 
+def filter_template(filename_pulse_shape, dt):
+
+    time_steps, amplitudes = np.loadtxt(filename_pulse_shape, unpack=True)
+    pulse_template = interp1d(time_steps, amplitudes, kind='cubic',
+                                   bounds_error=False,
+                                   fill_value=0., assume_sorted=True)
+
+    temp_times = np.arange(time_steps[0], time_steps[-1],
+                           (time_steps[1] - time_steps[0])/1000)
+
+    temp_amplitudes = pulse_template(temp_times)
+    amplitudes = amplitudes / np.max(temp_amplitudes)
+    pulse_template = interp1d(time_steps, amplitudes, kind='cubic',
+                                   bounds_error=False,
+                                   fill_value=0., assume_sorted=True)
+
+    t = np.arange(0, 92*4, 4)
```

Comment:
`dt` seems to be unused and these magic numbers should maybe be removed at some place ... 

I like the use of `scipy.interpolate.interp1d` much better than `from scipy.interpolate import splrep, splev` ... *thumbs up*

Reply:
fixed dt is passed as argument (as well as n_samples i.e. 92)
FYI we usually use 92 instead of 50 samples for in-lab calibration data